plugins/module_utils/vmware_rest_client.py

```diff
@@ -381,178 +220,121 @@ def get_datacenter_by_name(self, datacenter_name):
         Returns the identifier of a datacenter
         Note: The method assumes only one datacenter with the mentioned name.
         """
+        if datacenter_name is None:
+            return None
+
         filter_spec = Datacenter.FilterSpec(names=set([datacenter_name]))
         datacenter_summaries = self.api_client.vcenter.Datacenter.list(filter_spec)
-        datacenter = datacenter_summaries[0].datacenter if len(datacenter_summaries) > 0 else None
-        return datacenter
+        return datacenter_summaries[0].datacenter if len(datacenter_summaries) > 0 else None
 
-    def get_folder_by_name(self, datacenter_name, folder_name):
+    def get_folder_by_name(self, folder_name, datacenter_name=None):
         """
         Returns the identifier of a folder
         with the mentioned names.
         """
-        datacenter = self.get_datacenter_by_name(datacenter_name)
-        if not datacenter:
+        if folder_name is None:
             return None
+        datacenter = self.get_datacenter_by_name(datacenter_name)
+        datacenters = set([datacenter]) if datacenter else set()
         filter_spec = Folder.FilterSpec(type=Folder.Type.VIRTUAL_MACHINE,
                                         names=set([folder_name]),
-                                        datacenters=set([datacenter]))
+                                        datacenters=datacenters)
         folder_summaries = self.api_client.vcenter.Folder.list(filter_spec)
-        folder = folder_summaries[0].folder if len(folder_summaries) > 0 else None
-        return folder
+        return folder_summaries[0].folder if len(folder_summaries) > 0 else None
 
-    def get_resource_pool_by_name(self, datacenter_name, resourcepool_name, cluster_name=None, host_name=None):
+    def get_resource_pool_by_name(self, resourcepool_name, datacenter_name=None, cluster_name=None, host_name=None):
         """
         Returns the identifier of a resource pool
         with the mentioned names.
         """
         datacenter = self.get_datacenter_by_name(datacenter_name)
```

Comment:
those 2 lines :
 datacenter = self.get_datacenter_by_name(datacenter_name)
 datacenters = set([datacenter]) if datacenter else set()

 are repeated at  least 5 times - can you replace the function to be get_datasenter_set_by_names?